src/backend/distributed/utils/statistics_collection.c

```diff
@@ -86,8 +100,138 @@ CollectBasicUsageStatistics(void)
+		HTTP_TIMEOUT_SECONDS, &CheckForUpdatesCallback);
+}
+
+
+/*
+ * CheckForUpdatesCallback receives the response for the request sent by
+ * CheckForUpdates(). It processes the response, and if there is a newer release
+ * of Citus available, logs a LOG message. This function returns 0 if there are
+ * any errors in the received response, which means we didn't consume the data.
+ * Otherwise, it returns (size * count) which means we consumed all of the data.
+ */
+static size_t
+CheckForUpdatesCallback(char *contents, size_t size, size_t count, void *userData)
+{
+	PG_TRY();
+	{
+		const int citusVersionMajor = CITUS_VERSION_NUM / 10000;
+		const int citusVersionMinor = (CITUS_VERSION_NUM / 100) % 100;
+		const int citusVersionPatch = CITUS_VERSION_NUM % 100;
+		Datum responseJson = 0;
+		StringInfo releaseVersion = NULL;
+		int releaseMajor = 0;
+		int releaseMinor = 0;
+		int releasePatch = 0;
+		char *updateType = NULL;
+
+		StringInfo responseNullTerminated = makeStringInfo();
+		appendBinaryStringInfo(responseNullTerminated, contents, size * count);
+		responseJson = DirectFunctionCall1(jsonb_in,
+										   CStringGetDatum(responseNullTerminated->data));
+
+		if (!JsonbFieldExists(responseJson, "version") ||
+			!JsonbFieldExists(responseJson, "major") ||
+			!JsonbFieldExists(responseJson, "minor") ||
+			!JsonbFieldExists(responseJson, "patch"))
+		{
+			return 0;
+		}
+
+		releaseVersion = JsonbFieldStr(responseJson, "version");
+		releaseMajor = JsonbFieldInt(responseJson, "major");
+		releaseMinor = JsonbFieldInt(responseJson, "minor");
+		releasePatch = JsonbFieldInt(responseJson, "patch");
+
+		if (releaseMajor > citusVersionMajor)
+		{
+			updateType = "major";
+		}
+		else if (releaseMajor == citusVersionMajor &&
+				 releaseMinor > citusVersionMinor)
+		{
+			updateType = "minor";
+		}
+		else if (releaseMajor == citusVersionMajor &&
+				 releaseMinor == citusVersionMinor &&
+				 releasePatch > citusVersionPatch)
+		{
+			updateType = "patch";
+		}
+
+		if (updateType != NULL)
+		{
+			ereport(LOG, (errmsg("a newer %s release of Citus (%s) is available",
+								 updateType, releaseVersion->data)));
+		}
+
+		return size * count;
+	}
+	PG_CATCH();
+	{
+		FlushErrorState();
+		return 0;
+	}
+	PG_END_TRY();
+}
+
+
+/* JsonbFieldExists checks if the given field exists in the given JSONB object. */
+static bool
+JsonbFieldExists(Datum jsonb, const char *fieldName)
+{
+	return DatumGetBool(DirectFunctionCall2(jsonb_exists, jsonb,
+											CStringGetTextDatum(fieldName)));
+}
+
+
+/*
+ * JsonbFieldInt returns integer value of the given field in the given JSONB
+ * object. Caller should have checked that the field exists before calling this
+ * function. If field value cannot be converted to an integer, throws an error.
+ */
+static int
+JsonbFieldInt(Datum jsonb, const char *fieldName)
+{
+	StringInfo fieldStr = JsonbFieldStr(jsonb, fieldName);
+	return pg_atoi(fieldStr->data, fieldStr->len, '\0');
```

Comment:
This is super circuitous: `JsonbFieldStr` gets the underlying integer, then converts it to text, then we're making a C string, then we're calling `pg_atoi`. Is there some way to use `findJsonbValueFromContainer` to accomplish what we want here?